src/tests/nativeaot/SmokeTests/SharedLibrary/SharedLibrary.csproj

```diff
@@ -55,5 +56,11 @@ cp SharedLibraryDriver$(NativeExtension) native/SharedLibrary$(NativeExtension)
 
   <ItemGroup Condition="'$(WasiReactorTestUsingNodeJS)' == 'true'">
     <Content Include="SharedLibraryDriver.mjs" CopyToOutputDirectory="PreserveNewest" />
+    <NativeLibrary Include="$(MSBuildProjectDirectory)/LibraryWorld_component_type.o" />
   </ItemGroup>
+
+  <Target Name="CheckWasmSdks" Condition="'$(WasiReactorTestUsingNodeJS)' == 'true'">
+    <Error Text="WASI_SDK_PATH not set; please install WASI-SDK 22 and set WASI_SDK_PATH to the path where it is installed."
+           Condition="'$(WASI_SDK_PATH)' == ''" />
+  </Target>
```

Comment:
Needed?

Reply:
Maybe not?  I think I added it before I understood how `src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets` got injected into projects, and even now I'm not sure I understand it.  Can we rely on the `CheckWasmSdks` target in `Microsoft.NETCore.Native.targets` always being run implicitly as part of an application's build process?

Reply:
>  Can we rely on the CheckWasmSdks target in Microsoft.NETCore.Native.targets always being run implicitly as part of an application's build process?

Yep.

> and even now I'm not sure I understand it

For runtime tests such as this one, it is explicitly references through `Microsoft.DotNet.ILCompiler.SingleEntry.targets` in `src/tests/Directory.Build.targets`.

For 'normal' projects that reference the compiler via a NuGet package, NuGet targets inject `<package name>.props` and `<package name>.targets` from the `build` folder into the project (as part of restore, I believe).